src/discordcr/mappings/rest.cr

```diff
@@ -22,5 +29,37 @@ module Discord
         code: String
       )
     end
+
+    # A request payload to rearrange channels in a `Guild` by a REST API call.
+    struct ModifyChannelPositionPayload
+      def initialize(@id : UInt64, @position : Int32,
+                     @parent_id : UInt64 | MaybeField = MaybeField::Unchanged,
+                     @lock_permissions : Bool? = nil)
+      end
+
+      def to_json(builder : JSON::Builder)
+        builder.object do
+          builder.field("id") do
+            SnowflakeConverter.to_json(@id, builder)
+          end
+
+          builder.field("position", @position)
+
+          parent = @parent_id
+
+          if parent.is_a?(MaybeField)
+            if parent.none?
+              builder.field("parent_id", nil)
```

Comment:
Should I prefer a guard style here? I think that would be more consistent with the `lock_permissions` field `unless` guard below.